tests/src/tests/README.md

```diff
@@ -0,0 +1,231 @@
+        * ```func (devfile *TestDevfile) UpdateCommand(command *schema.Command) error```
+            * includes logic to call set<commad-type>CommandValues for each commandType.
+        * ```func (devfile TestDevfile) VerifyCommands(parserCommands []schema.Command) error```
+            * Includes logic to compare the array of commands obtained from the parser with those created by the test. if the compare fails:
+                * each individual command is compared.
+                    * if a command compare fails, the parser version and test version of the command are oputput as yaml files  to the tmp directory 
+                * a check is made to determine if the parser returned a command not known to the test or the pasrer omitted a command expected by the test.
+1. ```test-utils.go```
+    * ```func (devfile TestDevfile) Verify()``` 
+        * Includes code to get object from the paser and verify their content.
+        * For commands code is required to: 
+            1. Retrieve each command from the parser
+            1. Use command id to obtain the GenericCommand object which matches
+            1. compare the command structure returned by the parser with the command structure saved in the GenericCommand object.
+    * ```func (devfile TestDevfile) EditCommands() error```
+        * Specific to command objects.
+            1. Ensure devfile is written to disk
+            1. use parser to read devfile and get all command object
+            1.  for each command call:
+                *  ```func (devfile *TestDevfile) UpdateCommand(command *schema.Command) error``` 
+            1. when all commands have been updated, use parser to write the updated devfile to disk
+1. ```parser-v200-test.go```
+    * ```type TestContent struct```
+        * includes an array of command types: ```CommandTypes     []schema.CommandType``` 
+    * ```func Test_ExecCommand(t *testing.T)```
+        1. Creates a TestContent object
+        1. Adds a single entry array containg schema.ExecCommandType to the array of command types
+        1. Calls runTest for a single thread test
+        1. Calls runMultiThreadTest for a multi-thread test.
+    * See also
+        *```func Test_<string>ExecCommand(t *testing.T)``` 
+        *```func Test_MultiCommand(t *testing.T)```
+        *```func Test_Everything(t *testing.T)```
+    * add logic to ```func runTest(testContent TestContent, t *testing.T)``` includes logic
+        1. Add commands to the test
+        2. Starts edits of commands if required. 
+
+
+#### Code flow
+
+Create, modify and verify an exec command: 
+1. parser_v200_verify_test.Test_ExecCommand
+    1. parser-v200-test.runTest
+        1. command-test-utils.AddCommand
+            1. command-test-utils.GenerateCommand
+                1.  command-test-utils.createExecCommand
+                    1. command-test-utils.setExecCommandValues
```

Comment:
check the indentation, otherwise the format will be incorrect. 
same same below
```suggestion
        1. command-test-utils.AddCommand
        1. command-test-utils.GenerateCommand
        1.  command-test-utils.createExecCommand
        1. command-test-utils.setExecCommandValues
```